### PR TITLE
Simplified filter page

### PR DESCRIPTION
**Pull request by cailors**

### CODE REVIEW

**Comment by cailors** [comment]
Screen Shot 2022-08-01 at 8 04 26 PM

- Searching by "ab" will return items with name "ab cd" and "cd ab" but not "cdab"
- Filtering with rate 1 with return results with rating between 1 and 2

File: routes/items.js [hunk 1]
```diff
@@ -28,11 +28,10 @@ router.get('/', async (req, res) => {
   const { search, type, rating, status, location } = req.query
 
   const usersInLocation = location ? (await User.find({ location })).map((user) => user._id) : undefined
-
   const query = {
-    ...(search ? { name: { $regex: new RegExp(search, 'i') } } : {}),
+    ...(search ? {$or:[{name:{ $regex: new RegExp( "^"+search, 'i') }},{name:{ $regex: new RegExp( " "+search, 'i') }}]} : {}),
```
Comment (phreid) [inline routes/items.js]: Remember to use the linter on both the frontend and backend files.

**Review by phreid — CHANGES_REQUESTED** [review]
Looks good, just another linting issue.

File: routes/items.js (hunk 1 above)
Comment (cailors) [reply routes/items.js]: will look into it, thanks

**Comment by imogenem** [comment]
It took me some time to understand what the filter buttons do now, so maybe we should simplify this now. Since the "apply all" button applies the search and filters, I think we should remove the 'Clear Filters' button since "reset all" covers that functionality.

But otherwise looks good and works on my end.

**Comment by phreid** [comment]
> It took me some time to understand what the filter buttons do now, so maybe we should simplify this now. Since the "apply all" button applies the search and filters, I think we should remove the 'Clear Filters' button since "reset all" covers that functionality.
> 
> But otherwise looks good and works on my end.

Are these buttons being changed? Don't see it in here.

**Pull request merged**